app/repos/RepoDrawer.tsx

```diff
@@ -37,15 +45,46 @@ const RepoDrawer = () => {
   useEffect(() => {
     if (!session) return;
     if (repos.length > 0) return;
+    if (!session?.provider_token) return;
+
+    getPaginatedRepos(session?.provider_token)
+      .then((allRepos: any) => {
+        if (allRepos?.nodes?.length) {
+          setRepos(allRepos.nodes);
+          setReposCount(allRepos.totalCount);
+          setPageInfo(allRepos.pageInfo);
+        }
+      })
+      .catch((err: any) => {
+        console.log("Failed to get repos:", { err });
+      });
+  }, [repos.length, session, user]);
 
-    getRepos(session?.provider_token)
-      .then((allRepos) => {
-        setRepos(allRepos);
+  const onPreviousPage = async () =>
+    session?.provider_token &&
+    getPaginatedRepos(session?.provider_token, pageInfo?.endCursor)
+      .then((allRepos: any) => {
```

Comment:
just curious but why `any` ? TypeScript already know the return type from `getPaginatedRepos`